clients/osv.go

```diff
@@ -45,26 +46,32 @@ func (v osvClient) ListUnfixedVulnerabilities(
 		Recursive:      true,
 		GitCommits:     gitCommits,
 	}, nil) // TODO: Do logging?
-	if err != nil {
-		return VulnerabilitiesResponse{}, fmt.Errorf("osvscanner.DoScan: %w", err)
-	}
 
 	response := VulnerabilitiesResponse{}
-	vulns := res.Flatten()
-	for i := range vulns {
-		response.Vulnerabilities = append(response.Vulnerabilities, Vulnerability{
-			ID:      vulns[i].Vulnerability.ID,
-			Aliases: vulns[i].Vulnerability.Aliases,
-		})
-		// Remove duplicate vulnerability IDs for now as we don't report information
-		// on the source of each vulnerability yet, therefore having multiple identical
-		// vuln IDs might be confusing.
-		response.Vulnerabilities = removeDuplicate(
-			response.Vulnerabilities,
-			func(key Vulnerability) string { return key.ID },
-		)
+	if err == nil {
+		return response, nil
+	}
+
+	if errors.Is(err, osvscanner.VulnerabilitiesFoundErr) {
```

Comment:
should this be an `!errors.Is()`? I may be mis-understanding what `osvscanner.VulnerabilitiesFoundErr`. Can you add a comment? Does it mean that *some* results were populated, but not all?

Reply:
I don't believe so based on the code for `DoScan` https://github.com/google/osv-scanner/blob/6eab4504663395b4f5983ebe4570e8b816844ab5/pkg/osvscanner/osvscanner.go#L469-L479

Reply:
Added some additional comments to clarify it, Spencer is correct, if vulnerabilities are found VulnerabilitiesFoundErr will be returned.